### DISCUSSION TITLE
Subject Identifiers Using zero fill

### DISCUSSION THREAD

**Discussion by alexrockhill** [bids-standard/bids-specification#1679] [General]
Why do subject identifiers use zero fills as in here on the BIDS home page?

![image](https://github.com/bids-standard/bids-specification/assets/13473576/ced5a066-a8ea-4ffa-bc2b-689bcb20aafa)

Is this still the standard? This seems very old school as well strings were allocated a set number of bits.

Related to https://github.com/mne-tools/mne-bids/pull/1215.

Should `mne_bids` for instance be encouraging people or requiring people to use this kind of format? I would guess not but I was a bit surprised to see it so ubiquitously in BIDS and thought there must be a modern reason for this kind of coding.

**Comment by Remi-Gau** [comment]
https://bids-specification.readthedocs.io/en/latest/common-principles.html#participant-names-and-other-labels

> In indices we RECOMMEND using zero padding (for example, 01 instead of 1 if some participants have two-digit labels) to make alphabetical sorting more intuitive.

Adding to that it may facilitate some aspects of scripting if most filenames follow a more similar pattern.

Note that subject label do not need to be digits but if they are the above recommendation would apply.

In the example above, it may not be obvious as you have less than 10 subjects.

**Comment by alexrockhill** [comment]
Thanks for linking that part of the documentation, I hadn't seen that!

That seems like a half-decent explanation for why do to that although you can easily check if subject idenitifiers consist entirely of digits and if so cast to integers before sorting so I'm not sure that really convinces me that I should be using it especially when the downside is subject 100 wouldn't fit into the paradigm. I'm not opinionated that it shouldn't be used, I just think subject identifiers should be allowed to be digits per the discussion in the linked issue and wanted to make sure it wasn't against core BIDS principles. It sounds like because it's recommended, it's not bad to have that be in examples and defaults but it seems overly restrictive and not required by BIDS to require it for `mne_bids`.

**Reply by Remi-Gau** [reply]
I think that in this context sorting should also be understood more generally as how (for example) a file browser GUI (file explorer, finder, nautilus...) would display the files if you asked it to sort them by names.

Remember that BIDS is for machines AND humans. 😉 

Or even just a `tree` command.

you would end up with things like:

```
.
├── sub-1
├── sub-10
└── sub-2
```

**Comment by alexrockhill** [comment]
Sure, but we also have `print_tree` in `mne-bids` for instance and I'm sure other BIDS packages wrap similar commands and could sort them more logically for you. I agree with the idea and I think it's an easy hack for sorting but it still seems like a bit of a hack rather than a principled way to do things. Especially for a standard where subject numbers may vary by orders of magnitude so some people will use `01` and others `001` and maybe still others `0001`. I don't think it shouldn't be recommended, it's just not my personal preference.

**Reply by Remi-Gau** [reply]
> seems like a bit of a hack rather than a principled way to do things

sometimes it is just easier to get people to use a hack than all get them to agree on the principle

but as you say the good thing about BIDS is that it still leaves you with some flexibility for personal preferences, so I defo won't be runnning after you to make you zero pad your subject labels. 😉

**Reply by alexrockhill** [reply]
Haha thanks for the discussion!
